.github/workflows/main.yml

```diff
@@ -0,0 +1,106 @@
+name: Main
+
+on:
+  push:
+    branches:
+      - dynamo_py3
+  pull_request:
+
+jobs:
+  build-test:
+    name: Build and Test
+    runs-on: ${{ matrix.os.instance }}
+    timeout-minutes: 15
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - category: windows
+            platform: x86
+            instance: windows-latest
+
+          - category: windows
+            platform: x64
+            instance: windows-latest
+
+          - category: ubuntu
+            platform: x64
+            instance: ubuntu-22.04
+
+          - category: macos
+            platform: x64
+            instance: macos-13
+
+        python: ["3.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
+
+    steps:
+      - name: Set Environment on macOS
+        uses: maxim-lobanov/setup-xamarin@v1
+        if: ${{ matrix.os.category == 'macos' }}
+        with:
+          mono-version: latest
+
+      - name: Checkout code
+        uses: actions/checkout@v2
```

Comment:
Currently v4 is out, any reason to stick to v2?

Reply:
I just copied this from upstream, other than turning off the performance tests, which weren't working. I think it makes sense to match upstream, but what do you think?

Reply:
Got it, makes sense.